src/main/scala/com/cognite/sdk/scala/v1/dataModels.scala

```diff
@@ -3,81 +3,49 @@
 
 package com.cognite.sdk.scala.v1
 
+final case class SpacedItems[A](spaceExternalId: String, items: Seq[A])
+
+final case class DataModelIdentifier(
+    space: Option[String],
+    model: String
+)
+
 final case class DataModelProperty(
-    `type`: String,
+    `type`: PropertyType,
     nullable: Boolean = true,
-    targetModelExternalId: Option[String] = None
+    targetModel: Option[DataModelIdentifier] = None
 )
 
-final case class DataModelPropertyIndex(
-    indexName: Option[String] = None,
-    fields: Option[Seq[String]] = None
+final case class UniquenessConstraint(
+    uniqueProperties: Seq[String]
 )
 
-final case class DataModel(
-    externalId: String,
-    properties: Option[Map[String, DataModelProperty]] = None,
-    `extends`: Option[Seq[String]] = None,
-    indexes: Option[Seq[DataModelPropertyIndex]] = None
+final case class DataModelConstraints(
+    uniqueness: Option[Map[String, UniquenessConstraint]] = None
 )
 
-final case class DataModelListInput(includeInheritedProperties: Boolean)
-
-final case class DataModelGetByExternalIdsInput[A](
-    items: Seq[A],
-    includeInheritedProperties: Boolean,
-    ignoreUnknownIds: Boolean
+final case class BTreeIndex(
+    properties: Seq[String]
 )
 
-final case class DataModelInstanceCreate(
-    modelExternalId: String,
-    properties: Option[Map[String, PropertyType]] = None
+final case class DataModelIndexes(
+    btreeIndex: Option[Map[String, BTreeIndex]] = None
 )
 
-sealed trait DataModelInstanceFilter
+final case class DataModel(
+    externalId: String,
+    properties: Option[Map[String, DataModelProperty]] = None,
+    `extends`: Option[Seq[DataModelIdentifier]] = None,
+    indexes: Option[DataModelIndexes] = None,
+    constraints: Option[DataModelConstraints] = None,
+    instanceType: DataModelInstanceType = DataModelInstanceType.Node
```

Comment:
Last nit-pick, I promise: Let's rename it to dataModelType. DataModelInstance used to mean something specific, so let's purge it from the codebase.